google/cloud/pubsub/internal/subscriber_tracing_connection.cc

```diff
@@ -68,10 +68,13 @@ StatusOr<pubsub::PullResponse> EndPullSpan(
         /*sc::kMessagingMessageEnvelopeSize=*/"messaging.message.envelope.size",
         static_cast<std::int64_t>(MessageSize(message)));
 
+    auto current = opentelemetry::context::RuntimeContext::GetCurrent();
     auto context = ExtractTraceContext(message, *propagator);
     auto producer_span_context =
         opentelemetry::trace::GetSpan(context)->GetContext();
-    if (producer_span_context.IsSampled() && producer_span_context.IsValid()) {
+    // If the contexts are equal, the message span was invalid.
```

Comment:
Note to self:

I don't think the producer_span_context.IsValid() line checks anything, that's what the check darren is adding will now check.

If the current context is equal to the extracted context, it means that the context was not valid.